tests/foreman/ui/test_registration.py

```diff
@@ -121,3 +128,496 @@ def test_negative_global_registration_without_ak(
+    result = client.execute(cmd)
+    assert result.status == 0
+    result = client.execute('yum list installed | grep mlocate')
+    assert result.status == 0
+    assert 'mlocate' in result.stdout
+    result = client.execute(f'yum -v repolist {repo_name}')
+    assert result.status == 0
+    assert repo_url in result.stdout
+
+
+@pytest.mark.tier3
+@pytest.mark.usefixtures('enable_capsule_for_registration')
+def test_global_re_registration_host_with_force_ignore_error_options(
+    session, module_activation_key, default_os, default_smart_proxy, rhel7_contenthost
+):
+    """If the ignore_error and force checkbox is checked then registered host can
+    get re-registered without any error.
+
+    :id: 8f0ecc13-5d18-4adb-acf5-3f3276dccbb7
+
+    :expectedresults: Verify the force and ignore checkbox options
+
+    :CaseLevel: Integration
+
+    :steps:
+        1. create and sync repository
+        2. create the content view and activation-key
+        3. open the global registration form and select --force and --Ignore Errors option
+        4. registered the host with generated curl command
+        5. re-register the same host again and check it is getting registered
+
+    :parametrized: yes
+    """
+    client = rhel7_contenthost
+    with session:
+        cmd = session.host.get_register_command(
+            {
+                'general.operating_system': default_os.title,
+                'general.capsule': default_smart_proxy.name,
+                'general.activation_keys': module_activation_key.name,
+                'general.insecure': True,
+                'advanced.force': True,
+                'advanced.ignore_error': True,
+            }
+        )
+    client.execute(cmd)
+    result = client.execute('subscription-manager identity')
+    assert result.status == 0
+    # rerun the register command
+    client.execute(cmd)
+    result = client.execute('subscription-manager identity')
+    assert result.status == 0
+
+
+@pytest.mark.tier2
+@pytest.mark.usefixtures('enable_capsule_for_registration')
+def test_global_registration_token_restriction(
+    session, module_activation_key, rhel8_contenthost, default_os, default_smart_proxy, target_sat
+):
+    """Global registration token should be only used for registration call, it
+    should be restricted for any other api calls.
+
+    :id: 4528b5c6-0a6d-40cd-857a-68b76db2179b
+
+    :expectedresults: global registration token should be restricted for any api calls
+        other than the registration
+
+    :CaseLevel: Integration
+
+    :steps:
+        1. open the global registration form and generate the curl token
+        2. use that curl token to execute other api calls e.g. GET /hosts, /users
+
+    :parametrized: yes
+    """
+    client = rhel8_contenthost
+    with session:
+        cmd = session.host.get_register_command(
+            {
+                'general.operating_system': default_os.title,
+                'general.capsule': default_smart_proxy.name,
```

Comment:
```suggestion
```

Reply:
I am going to address all the updates in test in https://github.com/SatelliteQE/robottelo/pull/14001